src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -147,7 +149,14 @@ private enum StringificationType {
    */
   @AutoCodec
   static final class VectorArg {
-    private static final Interner<VectorArg> interner = BlazeInterners.newStrongInterner();
+    // The build always uses the same StarlarkSemantics instance during analysis.
+    private static final LoadingCache<StarlarkSemantics, Interner<VectorArg>> interner =
```

Comment:
I want to profile to see if the extra layer of cache lookup shows up (wanted to get a memory diff anyway). It seems the starlark semantics are only necessary if `map_each` is present, and as you said we only expect one instance anyway, so there are ways we can improve if necessary. Will get back to you.

Reply:
Or if my ideas are too difficult/awkward, use a null `StarlarkSemantics` if there's no `map_each`, and have a dedicated interner for those instances.

Reply:
I went with a different approach that introduces an optimized single entry cache implementation. What do you think about that?